src/utils/order_status_history_util.py

```diff
@@ -0,0 +1,72 @@
+from datetime import datetime
+from functools import reduce
+
+
+def _last_order_status_histories_list_to_map(
+    map_order_status_histories, last_order_status_history
+):
+    """
+    This function reduce a list of order status histories to a map.
+    """
+    map_order_status_histories[
+        last_order_status_history.order_id
+    ] = last_order_status_history
+    return map_order_status_histories
+
+
+def _update_to_status_and_to_time(
+    updated_last_status_histories,
+    order_status_history,
+    map_last_order_status_history,
+    updated_by,
+):
+    """
+    This function update the to_status and to_time of the last order status history taking the from_status and from
+    time of the new order status history of an order, using the order id.
+    """
+
+    if not map_last_order_status_history[order_status_history.order_id].to_status:
+        map_last_order_status_history[
+            order_status_history.order_id
+        ].updated_by = updated_by
+        map_last_order_status_history[
+            order_status_history.order_id
+        ].updated_date = datetime.now()
+        map_last_order_status_history[
+            order_status_history.order_id
+        ].to_status = order_status_history.from_status
+        map_last_order_status_history[
+            order_status_history.order_id
+        ].to_time = order_status_history.from_time
+        updated_last_status_histories.append(
+            map_last_order_status_history[order_status_history.order_id]
+        )
+    return updated_last_status_histories
+
+
+def update_last_order_status_history(
```

Comment:
Isn't it easier to do it in just one loop?

Reply:
done.